app/scripts/models/reliers/oauth.js

```diff
@@ -171,13 +177,24 @@ var OAuthRelier = Relier.extend({
     if (! this.get('email')) {
       this.set('email', this.get('loginHint'));
     }
+
     // OAuth reliers are not allowed to specify a service. `service`
     // is used in the verification flow, it'll be set to the `client_id`.
     if (this.getSearchParam('service')) {
       throw OAuthErrors.toInvalidParameterError('service');
     }
   },
 
+  _setupSuccessFlow () {
+    const pathname = this.window.location.pathname.split('/');
+    const clientId = pathname[pathname.length - 1];
+    if (! clientId) {
```

Comment:
Can you validate the clientId here to ensure it's legit?

Reply:
I Think the broker will validate it , non? or what do you mean about legit?

Reply:
The broker won't validate it, reliers validate.

Reply:
And by legit, I mean, ensure the thing is a 16 character hex code.

Reply:
Here's what happens if garbage is put in there:

<img width="1245" alt="Screenshot 2019-03-26 at 10 50 54" src="https://user-images.githubusercontent.com/848085/54991573-1fa39580-4fb5-11e9-8f6e-e3d060ba576b.png">

500 error trying to decode.

<img width="479" alt="Screenshot 2019-03-26 at 10 52 16" src="https://user-images.githubusercontent.com/848085/54991623-4235ae80-4fb5-11e9-915c-cd446792b389.png">

And here's what happens if the client_id is a string that doesn't match an actual client_id:

<img width="1167" alt="Screenshot 2019-03-26 at 10 51 05" src="https://user-images.githubusercontent.com/848085/54991595-2f22de80-4fb5-11e9-8b1a-6cfcd8ec26e4.png">

400, better, but we could avoid the network request by validating it's a 16 character hex code.

Reply:
Opened #7093